File: frontend/types/rag.ts
/** backend/app/schemas/rag.py 그대로. */

export interface RagChunk {
  chunk_id: string;
  source: string | null;
  topic: string | null;
  consultation_stage: string | null;
  region: string | null;
  excerpt: string;
  pii_removed: boolean;
  score: number | null;
}

export interface RagAnswerRequest {
  topic: string;
  question: string;
  region?: string;
  consultation_stage?: "계약전" | "계약중" | "사고후";
  top_k?: number;
}

export interface RagAnswerData {
  answer: string;
  is_mock: boolean;
  sources: RagChunk[];
  disclaimer: string;
  rag_search_log_id: string;
}

File: backend/app/schemas/rag.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class RagSearchRequest(BaseModel):
    topic: str = Field(min_length=1)
    region: str | None = None
    consultation_stage: Literal["계약전", "계약중", "사고후"] | None = None
    top_k: int = Field(default=3, ge=1, le=10)


class RagChunkResponse(BaseModel):
    chunk_id: str
    source: str | None = None
    topic: str | None = None
    consultation_stage: str | None = None
    region: str | None = None
    excerpt: str
    pii_removed: bool
    score: float | None = None


class RagSearchResponse(BaseModel):
    query: dict
    chunks: list[RagChunkResponse]
    is_mock: bool
    rag_search_log_id: str


class RagAnswerRequest(BaseModel):
    topic: str = Field(min_length=1)
    question: str = Field(min_length=1)
    region: str | None = None
    consultation_stage: Literal["계약전", "계약중", "사고후"] | None = None
    top_k: int = Field(default=3, ge=1, le=10)


class RagAnswerResponse(BaseModel):
    answer: str
    is_mock: bool
    sources: list[RagChunkResponse]
    disclaimer: str
    rag_search_log_id: str
